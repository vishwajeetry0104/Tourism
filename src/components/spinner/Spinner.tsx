import React from 'react';
import {ActivityIndicator, View, Modal, Text} from 'react-native';
import type {ViewStyle} from 'react-native';
import styles from './SpinnerStyle';
import { useIsFocused } from '@react-navigation/native';


type Props = {
  size?: number | 'small' | 'large';
  hidesWhenStopped?: boolean;
  animating?: boolean;
  inline?: boolean;
  isFocused?: boolean;
  useActivityIndicator?: boolean;
  onDismiss?: () => void;
  style?: ViewStyle;
  color?: string;
  title?: string;
  subTitle?: string;
  testID?: string;
};

const Spinner = (props: Props) => {
  const {
    hidesWhenStopped = true,
    animating = false,
    inline = false,
    isFocused = false,
    useActivityIndicator = true,
    onDismiss = () => {},
    style,
    color,
    title = 'Loading',
    subTitle,
    size = 'large',
    testID
  } = props;

  const isFocusedNavigation = useIsFocused()

  const {
    ActivityIndicatorStyle,
    ActivityIndicatorModalStyle,
    ActivityIndicatorColorStyle,
    titleStyles,
    subTitleStyles
  } = styles;

  const getLoader = () => (
    <>
      <ActivityIndicator
        testID={testID}
        size={size}
        animating={animating}
        hidesWhenStopped={hidesWhenStopped}
        color={color || ActivityIndicatorColorStyle?.color}
      />
      {title !== '' && (
        <Text style={titleStyles}>
          {title}
        </Text>
      )}
      <Text style={subTitleStyles}>
        {subTitle}
      </Text>
    </>
  );

  if (inline && animating) {
    return <View style={[ActivityIndicatorStyle, style]}>{getLoader()}</View>;
  }

  return (
    <Modal onDismiss={onDismiss} transparent visible={animating && (isFocused || isFocusedNavigation)}>
      <View style={ActivityIndicatorModalStyle}>{useActivityIndicator ? getLoader() : <></>}</View>
    </Modal>
  );
};

export default Spinner;
