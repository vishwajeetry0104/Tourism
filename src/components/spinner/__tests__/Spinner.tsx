import React from 'react';
import {act, create} from 'react-test-renderer';
import Spinner from '../Spinner';

describe('Spinner', () => {
  const props = {
    animating: true
  };

  const createWrapper = (updatedProps = props) => {
    return create(<Spinner {...updatedProps} />);
  };

  let defaultWrapper;

  act(() => {
    defaultWrapper = createWrapper();
  });

  afterAll(() => {
    defaultWrapper.unmount();
  });

  it('should render Spinner component without issues', () => {
    expect(defaultWrapper).toMatchSnapshot();
  });

  it('should render spinner with color', () => {
    const updatedProps = {
      ...props,
      color: '#000000'
    };
    let wrapper;
    act(() => {
      wrapper = createWrapper(updatedProps);
    });
    expect(wrapper).toMatchSnapshot();
  });
});
