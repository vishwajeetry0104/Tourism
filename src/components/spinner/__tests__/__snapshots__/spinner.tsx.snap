// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner should render Spinner component without issues 1`] = `
<Modal
  hardwareAccelerated={false}
  onDismiss={[Function]}
  transparent={true}
  visible={true}
>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "padding": 8,
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="#001A1A"
      hidesWhenStopped={true}
      size="large"
    />
    <Text
      style={
        {
          "color": "#001A1A",
          "fontSize": 20,
          "fontWeight": "700",
          "letterSpacing": 0.24,
          "paddingBottom": 8,
        }
      }
    >
      Loading
    </Text>
    <Text
      style={
        {
          "color": "#001A1A",
          "fontSize": 16,
          "fontWeight": "700",
          "letterSpacing": 0.24,
          "paddingBottom": 8,
        }
      }
    />
  </View>
</Modal>
`;

exports[`Spinner should render spinner with color 1`] = `
<Modal
  hardwareAccelerated={false}
  onDismiss={[Function]}
  transparent={true}
  visible={true}
>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "padding": 8,
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="#000000"
      hidesWhenStopped={true}
      size="large"
    />
    <Text
      style={
        {
          "color": "#001A1A",
          "fontSize": 20,
          "fontWeight": "700",
          "letterSpacing": 0.24,
          "paddingBottom": 8,
        }
      }
    >
      Loading
    </Text>
    <Text
      style={
        {
          "color": "#001A1A",
          "fontSize": 16,
          "fontWeight": "700",
          "letterSpacing": 0.24,
          "paddingBottom": 8,
        }
      }
    />
  </View>
</Modal>
`;
